modules/dxp/apps/commerce/commerce-shop-by-diagram/commerce-shop-by-diagram-web/src/main/resources/META-INF/resources/js/Diagram.js

```diff
@@ -0,0 +1,139 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
+
+import PropTypes from 'prop-types';
+import React, {useState} from 'react';
+
+import '../style/diagram.scss';
+import DiagramFooter from './DiagramFooter';
+import DiagramHeader from './DiagramHeader';
+import ImagePins from './ImagePins';
+
+const Diagram = ({
+	enablePanZoom,
+	enableResetZoom,
+	image,
+	imageSettings,
+	navigationController,
+	spritemap,
+	zoomController,
+}) => {
+	const [resetZoom, setResetZoom] = useState(false);
+	const [zoomInHandler, setZoomInHandler] = useState(false);
+	const [zoomOutHandler, setZoomOutHandler] = useState(false);
+	const [changedScale, setChangedScale] = useState(false);
+	const [scale, setScale] = useState(1);
+	const [selectedOption, setSelectedOption] = useState(1);
+
+	return (
+		<div className="diagram mx-auto">
+			<DiagramHeader setSelectedOption={setSelectedOption} />
+
+			<ImagePins
+				changedScale={changedScale}
+				enablePanZoom={enablePanZoom}
+				enableResetZoom={enableResetZoom}
+				image={image}
+				imageSettings={imageSettings}
+				navigationController={navigationController}
+				resetZoom={resetZoom}
+				scale={scale}
+				selectedOption={selectedOption}
+				setChangedScale={setChangedScale}
+				setResetZoom={setResetZoom}
+				setScale={setScale}
+				setSelectedOption={setSelectedOption}
+				setZoomInHandler={setZoomInHandler}
+				setZoomOutHandler={setZoomOutHandler}
+				zoomController={zoomController}
+				zoomInHandler={zoomInHandler}
+				zoomOutHandler={zoomOutHandler}
+			/>
+
+			<DiagramFooter
+				changedScale={changedScale}
+				enableResetZoom={enableResetZoom}
+				selectedOption={selectedOption}
+				setChangedScale={setChangedScale}
+				setResetZoom={setResetZoom}
+				setSelectedOption={setSelectedOption}
+				setZoomInHandler={setZoomInHandler}
+				setZoomOutHandler={setZoomOutHandler}
+				spritemap={spritemap}
+			/>
+		</div>
+	);
+};
+
+Diagram.defaultProps = {
+	enablePanZoom: true,
+	enableResetZoom: true,
+	image:
+		'https://i0.wp.com/detoxicrecenze.com/wp-content/uploads/2018/05/straight-6-engine-diagram-460-ford-engine-diagram-wiring-info-e280a2-of-straight-6-engine-diagram.jpg',
```

Comment:
Why not `imageUrl`? Also, is it a good practice adding external static resources Urls? @markocikos

Reply:
Sorry, this was an example just for testing purposes. The image url prop will be filled from a jsp with a media uploaded, ever as an (internal) url